### PR TITLE
Installer toolchain updates

### PR DESCRIPTION
**Pull request by L1ghtmann**
What does this implement/fix? Explain your changes.
---------------------------------------------------
- Adds aarch64 toolchain support
- Switches Swift toolchain to Kabir's 
- Updates toolchain deps
- Switches tar commands to curl | tar to remove need for intermediate files
- Fixes dnf 'update' command

Does this close any currently open issues?
------------------------------------------
No

Any relevant logs, error output, etc?
-------------------------------------

Any other comments?
-------------------
Swift OS releases > 5.8.1 appear to have a discrepancy with ABI reporting, so this will likely want to be the pinned version utilized until a fix is identified. Cc Leptos

Where has this been tested?
---------------------------
**Operating System:** …

Linux (WSL); Ubuntu 20.04, Ubuntu 22.04; SUSE Tumbleweed; SUSE Leap; Arch; Fedora 38; Debian

**Platform:** …

**Target Platform:** …

**Toolchain Version:** …

**SDK Version:** …

### CODE REVIEW

File: bin/install-theos
Comment (kabiroberai) [inline bin/install-theos:148]: nit: can we `-C` directly into `$THEOS/sdks` and remove the need for `$TMP` entirely?

Comment (leptos-null) [reply bin/install-theos:148]: I think it's helpful because the script only moves the `.sdk` items as opposed to all the scripts and other files in the archive

Comment (leptos-null) [reply bin/install-theos:148]: Just tested - using `tar xf - --strip=1 -C "${THEOS}/sdks/" "*.sdk"` worked well for me

File: bin/install-theos
Comment (leptos-null) [inline bin/install-theos:394]: why is this all commented out? should we just delete it?

File: bin/install-theos
Comment (leptos-null) [inline bin/install-theos:431]: it this the right link? both

```
https://github.com/L1ghtmann/llvm-project/releases/latest/download/iOSToolchain-aarch64.tar.xz
https://github.com/L1ghtmann/llvm-project/releases/latest/download/iOSToolchain-x86_64.tar.xz
```

are not found for me

Comment (leptos-null) [reply bin/install-theos:431]: never, I went to the repo listed and the most recent release just isn't published yet 👍

File: bin/install-theos
Comment (L1ghtmann) [reply bin/install-theos:394]: They're TBD. The Swift toolchain currently only supports Arch- and Debian-based distros.

File: bin/install-theos
Comment (L1ghtmann) [reply bin/install-theos:431]: Yup, want to wait for merge before publishing as the filenames are different and will break the currently revision of the installer which looks for the non-suffixed x64 file.

File: bin/install-theos
Comment (leptos-null) [reply bin/install-theos:148]:
```
tar: Option --wildcards is not supported
```

It appears the flag does not exist in BSD `tar`

File: bin/install-theos
Comment (leptos-null) [reply bin/install-theos:431]: ohh, makes sense. nice

File: bin/install-theos
Comment (leptos-null) [reply bin/install-theos:148]: While have less intermediates would be nice, perhaps more performance and even robust, given this discrepancy I feel more comfortable keeping the current `mktemp` and `mv` than trying to support both `tar` commands

Comment (L1ghtmann) [reply bin/install-theos:148]: Fun. How do we feel about a one-liner?
```
tar --version | grep GNU &> /dev/null && <tar cmd w/ --wildcards> || <tar cmd w/o --wildcards>
```

Comment (leptos-null) [reply bin/install-theos:148]: haha, no worries. that `tar --version` is fairly simple. I'm also good with either.

Attaching macOS 13.5.2 `tar` output for reference:

```
bsdtar 3.5.3 - libarchive 3.5.3 zlib/1.2.11 liblzma/5.0.5 bz2lib/1.0.8
```

Comment (L1ghtmann) [reply bin/install-theos]: Resolving as superseded by https://github.com/theos/theos/commit/1cfe85938ee26cac704f2e03f25ab114c5bfaa2f.

**Comment by L1ghtmann** [comment]
This is just about ready. Will need to publish my aarch64 toolchain, which I'm currently in the process of updating, and update Kabir's toolchain download links once the relevant prs land.

Recent changes include Swift RHEL support, the removal of Swift SUSE support for the time being, and the modification of the RHEL repository refresh cmd as the prior revision was non-functional (whoops). Swift toolchain deps should also be up-to-date.

**Comment by L1ghtmann** [comment]
In the interim while I work on a fix for the issues mentioned above, this pr has been modified to only adjust the following:
1) Introduces a new local bash variable `ARCH` to simplify some of the checks
2) Introduces the v2.3.0 Swift toolchain from Kabir to fix incompatibilities with older Clangs and iOS 15+ NSString format's
3) Switches tar commands to curl | tar to remove need for intermediate files
4) Fixes dnf 'update' command
5) Introduces aarch64 toolchains 
- Note: will need to push my release to latest once approved before merge in order for these to be available for download by the installer 
  
Requesting reviews when possible. Thanks!

**Review by leptos-null — APPROVED** [review]
Code looks good and simpler than before.
Tested on `ubuntu-latest` GitHub Actions runner.

**Pull request merged**
